src/cmdline/mod.rs

```diff
@@ -220,57 +168,51 @@ impl Cmdline {
                 .join(",")
         );
 
-        self.insert_str(kv_str)
+        self.insert_string(kv_str)
     }
 
-    /// Validates and inserts a string to the end of the current command line.
+    /// Validates and inserts a string to the end of the current args string.
     ///
     /// # Arguments
     ///
-    /// * `slug` - String to be appended to the command line.
-    ///
-    /// # Examples
-    ///
-    /// ```rust
-    /// # use linux_loader::cmdline::*;
-    /// # use std::ffi::CString;
-    /// let mut cl = Cmdline::new(100);
-    /// cl.insert_str("foobar");
-    /// let cl_cstring = CString::new(cl).unwrap();
-    /// assert_eq!(cl_cstring.to_str().unwrap(), "foobar");
-    /// ```
-    pub fn insert_str<T: AsRef<str>>(&mut self, slug: T) -> Result<()> {
-        let s = slug.as_ref();
+    /// * `slug` - String to be appended to the args string.
+    fn insert_string<T: AsRef<str>>(&mut self, slug: T) -> Result<()>;
+}
+
+impl UpdateArgs for String {
+    fn has_capacity(&self, more: usize) -> bool {
```

Comment:
Yes, that was what I was concerned in one of the comments above. From my point of view the solution to keep track of the size of the init args string while adding a new boot arg (and vice versa) would introduce a great complexity in the code that is not justified (I suppose we need to add a reference to the init string inside the boot arg structure and vice versa to be able to have the check properly implemented at this level, but the design would be hard to maintain).

However, my solution was to keep has_capacity from `Args` in order to prevent abuse (not to allow adding more that MAX_CAPACITY bytes into any of the init args string or boot args string because that would definitely result in an error when calling as_cstring()) and the real check already exist in the as_cstring().

Another option would be to have a method under `Cmdline` that takes two params (one specifying the new arguments that needs to be appended and the second one to specify the type of param that we try to add). This method can check if there is still space (according to all corner cases) because `Cmdline` has access to both init and boot arg strings and after that we call the corresponding insert method to the `init args` or `boot args` structs. In other words, to implement a wrapper over insert* methods in `Cmdline`. Doing like this mens we no longer need methods `boot_args_mut()` and `init_args_mut()` from `Cmdline`.

Would you like to go with this last approach?